include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
@Hypolite We need some check that the unserialize hadn't gone wrong. It returns "false" in that case - but the cache could return "false" as a valid value as well. But according to the documentation it raises a warning. So this could be used in some way. Since you are more familiar with PHP than @rabuzarus or me I thought of asking you for a clean solution.